crates/cheatcodes/src/inspector.rs

```diff
@@ -46,6 +47,63 @@ use std::{
     sync::Arc,
 };
 
+pub trait CheatcodesExecutor {
+    fn get_inspector<'a, DB: DatabaseExt>(
+        &'a mut self,
+        cheats: &'a mut Cheatcodes,
+    ) -> impl InspectorExt<DB> + 'a;
+
+    fn exec_create<DB: DatabaseExt>(
+        &mut self,
+        inputs: CreateInputs,
+        cheats: &mut Cheatcodes,
+        ecx: &mut InnerEvmContext<DB>,
+    ) -> Result<CreateOutcome, EVMError<DB::Error>> {
```

Comment:
this is quite clever, this invokes the evm with the regular inspector stack but as &mut.

this way we can also support multiple layers of evm calls, like vm.transact in a vm.transact